tools/nodejs/src/database.cpp

```diff
@@ -364,4 +366,31 @@ Napi::Value Database::RegisterReplacementScan(const Napi::CallbackInfo &info) {
 	return deferred.Promise();
 }
 
+Napi::Value Database::Tokenize(const Napi::CallbackInfo &info) {
+	auto env = info.Env();
+
+	if (info.Length() < 1) {
+		throw Napi::TypeError::New(env, "Text argument expected");
+	}
+
+	std::string text = info[0].As<Napi::String>();
+
+	auto tokens = duckdb::Parser::Tokenize(text);
```

Comment:
How well does this scale on huge queries?

Reply:
This just runs the scanner, not the full parser, so I assume the runtime is pretty close to linear in the input string.

Reply:
The underlying implementation is [here](https://github.com/duckdb/duckdb/blob/main/third_party/libpg_query/src_backend_parser_parser.cpp#L99).